Neos.Flow/Tests/Unit/Mvc/Controller/ActionControllerTest.php

```diff
@@ -246,6 +248,41 @@ public function processRequestInjectsSettingsToView()
         $this->actionController->processRequest($this->mockRequest, $mockResponse);
     }
 
+    public function supportedAndRequestedMediaTypes()
+    {
+        return [
+            // supported, Accept header, expected
+            [['application/json'], '*/*', 'application/json'],
+            [['text/html', 'application/json'], 'application/json', 'application/json'],
+            [['text/html'], 'text/html, application/xhtml+xml, application/xml;q=0.9, */*;q=0.8', 'text/html'],
+            [['application/json', 'application/xml'], 'text/html, application/json;q=0.7, application/xml;q=0.9', 'application/xml'],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider supportedAndRequestedMediaTypes
+     */
+    public function processRequestSetsNegotiatedContentTypeOnResponse($supportedMediaTypes, $acceptHeader, $expected)
+    {
+        $this->actionController = $this->getAccessibleMock(ActionController::class, ['resolveActionMethodName', 'initializeActionMethodArguments', 'initializeActionMethodValidators', 'resolveView', 'callActionMethod']);
+
+        $this->inject($this->actionController, 'objectManager', $this->mockObjectManager);
+
+        $mockMvcPropertyMappingConfigurationService = $this->createMock(Mvc\Controller\MvcPropertyMappingConfigurationService::class);
+        $this->inject($this->actionController, 'mvcPropertyMappingConfigurationService', $mockMvcPropertyMappingConfigurationService);
+
+        $mockHttpRequest = $this->getMockBuilder(ServerRequestInterface::class)->disableOriginalConstructor()->getMock();
+        $mockHttpRequest->method('getHeaderLine')->with('Accept')->willReturn($acceptHeader);
+        $this->mockRequest->method('getHttpRequest')->willReturn($mockHttpRequest);
+
+        $mockResponse = new Mvc\ActionResponse;
+        $this->inject($this->actionController, 'supportedMediaTypes', $supportedMediaTypes);
+
+        $this->actionController->processRequest($this->mockRequest, $mockResponse);
+        self::assertSame($expected, $mockResponse->getContentType());
```

Comment:
FYI @bwaidelich - we can't add a "does not set" case (supported != accepted) here, because `getContentType()` would try to return null and blow up - let's move the further discussion about the broken getter to https://github.com/neos/flow-development-collection/pull/2458